src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AppleCCCryptor.cs

```diff
@@ -0,0 +1,248 @@
+                    output.Length - outputOffset,
+                    out bytesWritten,
+                    out errorCode);
+            }
+
+            Exception exception = ProcessInteropError(ret, errorCode);
+
+            if (exception != null)
+            {
+                throw exception;
+            }
+
+            return bytesWritten;
+        }
+
+        private unsafe void OpenCryptor(
+            Interop.AppleCrypto.PAL_SymmetricAlgorithm algorithm,
+            CipherMode cipherMode,
+            byte[] key)
+        {
+            int ret;
+            int errorCode;
+
+            byte[] iv = IV;
+
+            fixed (byte* pbKey = key)
+            fixed (byte* pbIv = iv)
+            {
+                ret = Interop.AppleCrypto.CryptorCreate(
+                    _encrypting
+                        ? Interop.AppleCrypto.PAL_SymmetricOperation.Encrypt
+                        : Interop.AppleCrypto.PAL_SymmetricOperation.Decrypt,
+                    algorithm,
+                    GetPalChainMode(cipherMode),
+                    Interop.AppleCrypto.PAL_PaddingMode.None,
+                    pbKey,
+                    key.Length,
+                    pbIv,
+                    Interop.AppleCrypto.PAL_SymmetricOptions.None,
+                    out _cryptor,
+                    out errorCode);
+            }
+
+            Exception exception = ProcessInteropError(ret, errorCode);
+
+            if (exception != null)
+            {
+                throw exception;
+            }
+        }
+
+        private Interop.AppleCrypto.PAL_ChainingMode GetPalChainMode(CipherMode cipherMode)
+        {
+            switch (cipherMode)
+            {
+                case CipherMode.CBC:
+                    return Interop.AppleCrypto.PAL_ChainingMode.CBC;
+                case CipherMode.ECB:
+                    return Interop.AppleCrypto.PAL_ChainingMode.ECB;
+                case CipherMode.CTS:
+                    throw new PlatformNotSupportedException(SR.Cryptography_UnsupportedPaddingMode);
+                default:
+                    Debug.Fail($"Unknown cipher mode: {cipherMode}");
+                    throw new CryptographicException(SR.Cryptography_UnknownPaddingMode);
+            }
+        }
+
+        private unsafe void Reset()
+        {
+            int ret;
+            int errorCode;
+
+            byte[] iv = IV;
+
+            fixed (byte* pbIv = iv)
+            {
+                ret = Interop.AppleCrypto.CryptorReset(_cryptor, pbIv, out errorCode);
+            }
+
+            Exception exception = ProcessInteropError(ret, errorCode);
+
+            if (exception != null)
+            {
+                throw exception;
+            }
+        }
+
+        private static Exception ProcessInteropError(int functionReturnCode, int emittedErrorCode)
+        {
+            // Success
+            if (functionReturnCode == 1)
+            {
+                return null;
```

Comment:
It looks like everywhere this function is called, it's immediately followed by a check for null and throwing if the result is not null.  Rather than returning an Exception, why not just throw the exception from this method and return void?